src/common/operator/cast_operators.cpp

```diff
@@ -1409,7 +1409,11 @@ bool TryCastToTimestampNS::Operation(string_t input, timestamp_t &result, bool s
 	if (!TryCast::Operation<string_t, timestamp_t>(input, result, strict)) {
 		return false;
 	}
-	result = Timestamp::GetEpochNanoSeconds(result);
+	if (!Timestamp::IsFinite(result)) {
+		return true;
+	}
+	result = Timestamp::GetEpochNanoSeconds(
+	    result, StringUtil::Format("Could not convert VARCHAR value '%s' to Timestamp(NS)", input.GetString()));
```

Comment:
This looks to me like it is running a very expensive string format for every single value that is cast - can we do this in a manner that does not do this? Could you perhaps also run a benchmark to verify that this is actually very expensive?